pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -405,6 +408,48 @@ func (configgen *ConfigGeneratorImpl) buildGatewayInboundHTTPRouteConfig(
 	return out
 }
 
+// mergeIngress will append rules from Ingress-generated virtual services to existing virtual services.
+// The ingress can be created dynamically, and only has very limitted capabilities -
+// [host]/path->backend configuration
+func mergeIngress(configs []model.Config) []model.Config {
+	configsOut := []model.Config{}
+	hostsToVS := map[string]*networking.VirtualService{}
+
+	// Ignore ingress resources, build a map of hosts to (real) VirtualServices.
+	for _, c := range configs {
+		vs := c.Spec.(*networking.VirtualService)
+		for _, h := range vs.Hosts {
+			if !strings.HasSuffix(c.Name, model.IstioIngressGatewayName) {
```

Comment:
This is deduplicating for non ingress. Why?

Reply:
Sorry, mixed separate fix - it seems we still generate duplicate domains and RDS rejects it. 
I can send you the log.

I can remove this part of the PR and move it to separate.